src/states/ConfigMotionState.cpp

```diff
@@ -161,7 +167,13 @@ bool ConfigMotionState::run(mc_control::fsm::Controller &)
     }
   }
 
-  return !ctl().limbManagerSet_->contactCommandStacked() && taskConfigList_.empty() && collisionConfigList_.empty();
+  bool limbCofingsFinshed = !ctl().limbManagerSet_->contactCommandStacked();
+  if(exitWhenLimbSwingFinished_)
+  {
+    limbCofingsFinshed = limbCofingsFinshed && !ctl().limbManagerSet_->isExecutingLimbSwing();
+  }
+
+  return limbCofingsFinshed && taskConfigList_.empty() && collisionConfigList_.empty();
```

Comment:
```suggestion
  return !ctl().limbManagerSet_->contactCommandStacked() && taskConfigList_.empty() && collisionConfigList_.empty() && (!exitWhenLimbSwingFinished_ || !ctl().limbManagerSet_->isExecutingLimbSwing());
```
(should look nicer after formatting)

Reply:
Thank you for review. I refactored it.